api/types/provisioning.go

```diff
@@ -62,21 +62,39 @@ type ProvisionToken interface {
 	// GetSuggestedLabels returns the set of labels that the resource should add when adding itself to the cluster
 	GetSuggestedLabels() Labels
 
-	// V1 returns V1 version of the resource
-	V1() *ProvisionTokenV1
 	// String returns user friendly representation of the resource
 	String() string
+
+	// V3 returns the V3 representation of a ProvisionToken
+	// This exists for the transition period where V2 tokens can still be
+	// unmarshalled as V2 from the backend - and we need to convert them to
+	// V3 to send them over the wire to the client.
+	// This can be removed once we begin converting V2 tokens to V3 within the
+	// services.UnmarshalProvisionToken method.
+	V3() *ProvisionTokenV3
+
+	// V2 returns the V2 representation of a ProvisionToken
+	// This exists for the transition period where the V2 Token RPCs still exist
+	// and we may need to be able to convert tokens to a V2 representation
+	// for serialization.
+	//
+	// The second return value is a boolean indicating if this token can be
+	// represented as v2, for cases where a new provider type is used, this
+	// value will be false and the token should not be presented to a consumer.
+	V2() (*ProvisionTokenV2, bool)
```

Comment:
What would you say about replacing the `bool` with an `error`? I think that API would be easier to understand than a boolean.